File: go/gin/internal/modules/transaction/controller.go
package transaction

import (
	"net/http"
	"tcc/internal/common/errors"
	"tcc/internal/common/helper"
	"tcc/internal/common/messages"
	"tcc/internal/modules/transaction/dto"

	"github.com/gin-gonic/gin"
)

type TransactionController struct{
	service TransactionService
}

func (c *TransactionController) FindAll(ctx *gin.Context){
	idConta := ctx.Query("idConta")
	
	auth := helper.GetAuth(ctx)

	dto := dto.FindAllQueryTransactionsDto{
		IDConta: &idConta,
	}

	if err:= helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		ctx.JSON(badRequest.StatusCode, badRequest)
		return 
	}


	transactions, err := c.service.FindAll(*dto.IDConta, auth.ID, auth.Role)
	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	ctx.JSON(http.StatusOK, transactions)
}

func (c *TransactionController) Create(ctx *gin.Context){
	auth := helper.GetAuth(ctx)

	var dto dto.CreateTransactionDto
	if err := ctx.ShouldBindJSON(&dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err.Error())
		ctx.JSON(badRequest.StatusCode, badRequest)
		return
	}
	
	if err:= helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		ctx.JSON(badRequest.StatusCode, badRequest)
		return 
	}


	_, err := c.service.Create(dto, auth.ID)

	if err != nil {
		ctx.JSON(err.StatusCode, err)
		return
	}

	ctx.Status(http.StatusCreated)
}

File: rust/actix/src/modules/transaction/controller.rs
use actix_web::{
    web::{Json, Query},
    HttpResponse
};
use validator::Validate;
use crate::modules::auth::interfaces::JwtPayload;
use crate::modules::transaction::service::TransactionService;
use crate::modules::transaction::dto::CreateTransactionDto;
use crate::modules::transaction::dto::FindAllQueryTransactionsDto;
use crate::common::errors::bad_request::bad_request_error;
use crate::common::messages::error::ErrorMessages;
use crate::common::errors::base::BaseError;

use super::entity::TransactionEntity;
pub struct TransactionController;

impl TransactionController {
    pub async fn find_all(
        auth: JwtPayload,
        query: Query<FindAllQueryTransactionsDto>,
    ) -> Result<Json<Vec<TransactionEntity>>, BaseError> {
        match TransactionService::find_all(query.id_conta, auth.id, auth.role).await {
            Ok(transactions) => Ok(Json(transactions)),
            Err(err) => Err(err),
        }
    }

    pub async fn create(
        auth: JwtPayload,
        dto: Json<CreateTransactionDto>,
    ) -> Result<HttpResponse, BaseError> {
        if let Err(err) = dto.validate() {
            return Err(bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)));
        }

        if let Err(err) = dto.validate_transaction() {
            return Err(bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)));
        }

        match TransactionService::create(dto.into_inner(), auth.id).await {
            Ok(_) => Ok(HttpResponse::Created().finish()),
            Err(err) => Err(err),
        }
    }
} 